### PR TITLE
Removes VR

### LINKED ISSUE #37596: Blood-Crawl is kept outside of VR arena
If you obtain blood-crawl by killing a blood-demon in VR (put on by admins or the like), the effect is kept between VR charaters and outside of the arena (on the normal station).

```Round ID:  87255 

Server revision compiled on:  2018-04-30 

The following pull requests are currently test merged: 
#37587: 'Makes the jukebox less terrible' by KorPhaeron at commit 7ec266 
#34086: 'goonchat linkify has been vastly improved' by optimumtact at commit cd19a3 
#37406: 'Adds (fake) ambient occlusion as a client preference' by deathride58 at commit 87a9b1 


Based off origin/master commit: 21de54 

Revision: 74fd305540cef40eb4d3d480afe9efb65d69ad93 

Server tools version: /tg/station 13 Server v3.2.2.0 

Server tools API version: 3.2.0.2 

Current Informational Settings: ```

### LINKED ISSUE #38410: If you load snowdin VR and then snowdin (possibly when snowdin loads after anything?) the server hangs with the whitescreen
On a local server:

https://pastebin.com/Ax42Mv0C
https://i.imgur.com/h19ORO4.png

### LINKED ISSUE #39516: VR Holocalling
[Round ID]: 91910

[Reproduction]: Go into VR, call someone using a holopad, leave the call. Static.

image

### LINKED ISSUE #39869: why is stuff from VR giving me admin logs
help
https://i.gyazo.com/d0183ac53091a28add7c0c07c61e5aa7.png

### LINKED ISSUE #39946: Cannot Aghost while in VR
You get the admin message in chat but you stay trapped in your body, 

[2018-08-27 15:48:02.178] runtime error: Ghostize was called on a virtual reality mob
[2018-08-27 15:48:02.178] proc name: stack trace (/datum/proc/stack_trace)
[2018-08-27 15:48:02.178]   source file: unsorted.dm,1171
[2018-08-27 15:48:02.178]   usr: (src)
[2018-08-27 15:48:02.178]   src: Honkerlad (/mob/living/carbon/human/virtual_reality)
[2018-08-27 15:48:02.178]   src.loc: the floor (42,228,13) (/turf/open/floor/plasteel)
[2018-08-27 15:48:02.178]   call stack:
[2018-08-27 15:48:02.178] Honkerlad (/mob/living/carbon/human/virtual_reality): stack trace("Ghostize was called on a virtu...")

this is one of the many bugs VR has,such as the thousands of runtimes related to air, its loading being unified with away missions loading, etcetera.

### LINKED ISSUE #40079: Viscerators do nothing in syndie VR
Syndie VR players have the syndicate faction - any viscerators they release just fly around idly without attacking anyone.

### LINKED ISSUE #41508: Virtual reality entrepreneur discovers secret to fortune making
You can acquire free cash from the syndicate training VR mission.

Round ID: 97084

**Reproduction:**
Go to the syndicate training VR mission
Link the agent ID you're given to your actual station account
Buy loads of cash briefcases for the great deal of 1 TC each
Add cash to your ID, you are now rich
Leave VR and check your ID, you're still rich
...
(Stop here if you don't care about the rest of the story)
...
Let the power of cash go to your head
Claim you're a "virtual reality entrepreneur" should anyone question your fortune.
Buy out Cargonia
Bribe security to work for you instead
Use your bribed security to overthrow the captain
Worry about Nanotrasen finding your secret
Attempt to purchase station independence from Nanotrasen before they get to the truth
Realize Nanotrasen doesn't like anyone jeopardizing their monopoly on plasma
Become Nanotrasen's next target
Fund the Syndicate to provide protection
Escalate an already tense war between the Syndicate and Nanotrasen
Question your place in all this
Conclude that the FUN wasn't worth it
Wish you could turn back the clock
Pay for illegal bluespace chrono tech
Go back in time
Make this issue report
Go back to your job on SS13
Get bored
Look for more FUN
Repeat the vicious cycle
...
[No FUN allowed](https://tgstation13.org/wiki/Fun)

### LINKED ISSUE #43521: Plasma men are still plasma men in VR
If you are a plasma man, when you enter VR you remain as a plasma man but without  a plasma suit and plasma to breathe, you burn to death.

This makes VR a hellish death simulator where unknowing plasma men experience what it is like to burn and choke to death in toxic oxygen.

### LINKED ISSUE #43690: VR Hub pacifism carries over in Snowdin
## Round ID:
106786

## Testmerges:

Server revision compiled on: 
Master commit: 5239d718d6f088e8f4270c02c643e59e993063ac
The following pull requests are currently test merged:
#43349: 'Handle OoM errors.' by MrStonedOne at commit caf2ef8729

## Reproduction:
Enter VR, notice you're pacifist in most of the hub. Most other VR locations lift pacifism when appropriate, but you'll remain pacifist in VR Snowdin.

### LINKED ISSUE #43695: VR spawner has infinite respawns
Issue reported from  Round ID: 106849 (/tg/Station Bagil [ENGLISH] [US-WEST] [100% FREE LAG])

Reporting client version: 512.1462

You can infinitely respawn and ghost. I'll let the pictures speak for themselves.

image

image

This congaline continued to the spawnpoint.

### LINKED ISSUE #43719: Default cam on SS13 Network sends to vr
Pretty self explanatory, it should be obvious what the most worrisome about this is right now. Screenshot for evidence.

image

### LINKED ISSUE #43788: A thing about vr bossrush
## Round ID: 107417

Final room of bossrush has a shambling miner mob in it, and its also affected by pacifism. good thing it can be solved by book of firebolt there (also firebolt is not affected by pacifism)

## Reproduction:

step 1-beat bossrush
step 2-get into final room
step 3- get constantly bashed by a mob and cant retaliate becausue pacifism (atleast untill you learn firebolt)

### LINKED ISSUE #43842: Disgruntled entrepreneur strikes back, stealing billions of credits worth of spacecoin
Title, Protocal-17 CRAB phones can be bought in the Syndicate VR trainer uplinks.
## Round ID: 107835

MONEY
DUMPEET
## Reproduction:

- Go to syndicate trainer in virtual reality
- Open uplink and view devices menu
- > "Fudge what is this doing here?"
- Buy CRAB phones, cause repeated spacecoin market crashes until the system-wide market collapses
...
- Ruin another timeline as galactic civilization spirals into economic chaos
- We already know how this vicious cycle works
- You didn't listen
- You could have avoided this
- It's already too late
- it's happening
- Now we have to start over again.
...
[No.
FUN.
Allowed.](https://tgstation13.org/wiki/Stop)

Technically just another issue to be closed if #43832 is merged, though I don't enjoy providing another reason to have VR removed.

> LOOK SIR, FREE CRABS!

### LINKED ISSUE #43852: you can PDA people as a person that joined VR as a ghost
## Round ID: 
N/A

## Testmerges:
N/A

## Reproduction:
use the VR hub ghost spawner, goto the megafauna arena and use your PDA to message people on station

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Closes #43788
Closes #43719
Closes #43695
Closes #43690
Closes #43521
Closes #39946
Closes #39869
Closes #39516
Closes #43521
Closes #41508
Closes #40079
Closes #38410
Closes #37596
Closes #43842
Closes #43852

Also fixes an unreported issues with mindswaps into/out from VR and mind transfers such as your VR mob getting monkey'd killing you through VR.

## About The Pull Request

This removes VR. A better way to realize a separate VR dimension that cannot interact with the station is to run it as a separate server. That way the non-VR side of the server does not pay the performance cost of VR existing and people won't have their rounds ruined by shit interacting across the VR veil.

## Why It's Good For The Game

1. Most of it is singleplayer content that takes the player out of the round they were participating in, into a singleplayer instance we pay the cost of simulating atmos in
2. It's buggy as all hell and its intended design cannot be realized in a bug-free fashion without including "is this shit happening inside VR" checks everywhere, such as reagent code.

## Changelog
:cl: Naksu
del: Removed VR
/:cl:

### CODE REVIEW

**Comment by KetasDotepasto** [comment]
Well, one way to go about it, certainly.

**Comment by Whoneedspacee** [comment]
ouch

**Comment by Fikou** [comment]
oof

**Comment by SouDescolado** [comment]
I dunno how you would die because of VR but this feels pretty much an "i ded" issue

**Comment by HomocidlePancakes** [comment]
*insert that laughing newspaper guy from spiderman here*

**Comment by RaveRadbury** [comment]
The idea of putting VR on its own server sounds incredibly cool and would further add to the "hub" nature of VR. It would also pool the number of users which would likely increase its usage. Especially if we could get the Action Screens to show VR stuff. We could literally have cyber athletes from opposing stations.

**Comment by ATH1909** [comment]
https://i.kym-cdn.com/entries/icons/original/000/028/961/cover6.jpg

> The idea of putting VR on its own server sounds incredibly cool and would further add to the "hub" nature of VR. It would also pool the number of users which would likely increase its usage. Especially if we could get the Action Screens to show VR stuff. We could literally have cyber athletes from opposing stations.
> 
> **Event Hall for VR SERVER???**

This sounds rad, though, and I'd be down for it being a thing (if someone was willing to code it, of course).

File: code/modules/awaymissions/mission_code/mining.dm
Comment (chesse20) [inline code/modules/awaymissions/mission_code/mining.dm]: this can still be used outside of vr as a lootbox, keep it in

File: code/modules/awaymissions/mission_code/mining.dm
Comment (chesse20) [inline code/modules/awaymissions/mission_code/mining.dm]: this one too

**Review by chesse20 — CHANGES_REQUESTED** [review]
~~theres some good items in here also vr is fine.~~

**cobby note**: you do not need to input anything here when reviewing code if you don't have anything worthwhile.

**Comment by subject217** [comment]
>I dunno how you would die because of VR but this feels pretty much an "i ded" issue

the point is actually that there is a real and significant perf cost to run VR, and there's no working around that, and VR as is has intrinsic issues. you can't just scream "i ded!" at every feature removal you don't like.

**Comment by optimumtact** [comment]
putting it on it's own server is not even close to the same as what is offered here.

good technical reasons aside there's no way I can merge this

**Comment by actioninja** [comment]
> putting it on it's own server is not even close to the same as what is offered here.

How so?
Have VR sleepers shunt players to the VR server, and vice versa.

**Comment by Tlaltecuhtli** [comment]
remove snowdin and make the  tiles unsimulated because there is no reason for them to

**Comment by vuonojenmustaturska** [comment]
Putting it on its own server would have been a correct way to do what VR wants to do, and the only way to have a VR without side-effects into the current round. 

Having VR as-is in the codebase means every feature must be reviewed and tested for VR bugs, and coded with checks for "but what if this shit happens in VR" in place. Otherwise the new feature **will** introduce new potentially gameruining bugs, and this will happen forever and be the new normal.

This PR doesn't move VR into its own server for the very simple reason that I don't run tg, but also for the other reason that I don't believe in exclusive singleplayer content like this in what is a social multiplayer game.

**Comment by Barhandar** [comment]
Remove loot from snowdin and make it into lavaland's icecap, problem solved.

**Comment by nero1024** [comment]
How would the separate VR server even work, synchronize, and integrate with the normal game server? Is such a thing even feasible in Byond?

**Comment by vuonojenmustaturska** [comment]
it wouldn't have to, you can just ship clients to it and stop caring about them

**Comment by OverlyGeneric** [comment]
I have nothing to say about this. I'm flabbergasted this is even a thing.

**Comment by RaveRadbury** [comment]
Can we get some numbers posted so people can understand why this PR was made?

**Comment by bgobandit** [comment]
does anyone even use VR

**Comment by Qustinnus** [comment]
upvotes: the people who have to deal with this issue

downvotes: the people that play

is this the opposite of coders dont play?

**Comment by Borisvanmemes** [comment]
@Qustinnus Players don't code.

**Comment by Whoneedspacee** [comment]
vr really has no point once you move it to another server. i don’t think it’s much different than having a server with respawns enabled.

i’m mixed on removing it or not. on one hand i’ve seen it be very fun for players and i made most of the new stuff. on the other hand yeah it is almost more trouble than it’s worth and now that all these bugs are out even if you revert to old vr most of them will still be there now that people know about them.

i’d really be fine either way to be honest haha.

**Comment by nicbn** [comment]
Replace Sybil server with VR

**Comment by anconfuzedrock** [comment]
I'm sure there are plenty of other things with more issue reports than VR.

**Comment by OverlyGeneric** [comment]
> I'm sure there are plenty of other things with more issue reports than VR.

and at least vr is getting fixes for them.

**Comment by vuonojenmustaturska** [comment]
fuck

**Comment by vuonojenmustaturska** [comment]
oh yeah that works the commit is just a bit dirty

**Comment by FloranOtten** [comment]
How about we get plans to improve VR (Different server or no) before we remove VR

Improve, don't remove.

**Comment by PKPenguin321** [comment]
add actual VR support so i can truly be immersed

**Comment by ExcessiveUseOfCobblestone** [comment]
remove, have someone improve then re-pr is a better option :^)

**Comment by datorangebottle** [comment]
Oranges already said he won't merge this, so it's pretty dead.

Why does VR even bother simulating atmos instead of giving people a mob that doesn't need to breathe and is pressure proof? Barring one niche situation that could easily be removed(re: the magic mirrors in the hub) and aliens(who probably wouldn't be properly represented by a racist coder // by nanotrasen anyway).

**Comment by optimumtact** [comment]
actually I said there's no way I could merge it, but that doesn't imply I won't merge it

**Comment by optimumtact** [comment]
or that another maintainer wont

**Comment by ExcessiveUseOfCobblestone** [comment]
tempting

**Comment by optimumtact** [comment]
#43906 too

**Comment by optimumtact** [comment]
Given the balance of issues with this feature and even with recognising that this brings quite a few gameplay benefits for perma/ninja holding cell etc I will probably merge this, because on the balance, it's only a minor side feature as part of the larger game.

I am not opposed to the maps and hub returning later to be used on an tutorial/training server.

**Comment by vuonojenmustaturska** [comment]
if anyone decides to merge this, might want to re-run travis before you do just in case

**Comment by loserXD** [comment]
rest in piece

**Comment by mikolaj007** [comment]
F

**Comment by Nickkoranda** [comment]
f

**Pull request merged**
